libvips/foreign/vips2tiff.c

```diff
@@ -1188,23 +1190,32 @@ wtiff_new( VipsImage *input, const char *filename,
 		}
 	}
 
-	/* Can only squash 8 bit mono. 3-band float should have been squashed
-	 * above.
+	if( wtiff->bitdepth && !(wtiff->bitdepth == 1 || wtiff->bitdepth == 2 
+		|| wtiff->bitdepth == 4) ) {
+			g_warning( "%s",
+                _( "allows only bitdepth values 1,2 or 4. "
+					"-- disabling bitdepth") );
+            wtiff->bitdepth = 0;
+	}
+	
+    /* Can only have byte fractional bit depths for 8 bit mono.
+     * 3-band float should have been packed above.
 	 */
-	if( wtiff->squash &&
-		!(wtiff->ready->Coding == VIPS_CODING_NONE && 
+	if( wtiff->bitdepth && (wtiff->bitdepth == 1 || wtiff->bitdepth == 2 
+		|| wtiff->bitdepth == 4 )
+        && !(wtiff->ready->Coding == VIPS_CODING_NONE &&
```

Comment:
I'd just use `wtiff->bitdepth` here, you've checked the values in the test above.